  Of schools with 7000 or more undergraduate students, sorted descendingly by the % of students receiving federal loans..
  
  1.
  
    a. How many of the top 50 are located in NC?
    
    b. How many of the top 50 are distance only?
    
    c. What is the mean earning of students 10years after enrollment 
    of those top 50 schools?
    
  2. What is the national median debt for graduates (GRAD_DEBT_MDN_SUPP) compared to the median for NC schools?
    

Helpful notes!

Sorting by values
    https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.sort_values.html#pandas.DataFrame.sort_values
    
Documentation explaining column values    
https://collegescorecard.ed.gov/assets/FullDataDocumentation.pdf

For part C and #2, the data is being stored as objects (Strings) rather than numbers. This is because some schools have not provided that information, and it was entered as "PrivacySuppressed". You'll have to figure out a way around this.
    
    

# Introductory Commands

These were the commands used in lecture to help introduce the colleges and narrow down the dataframes.

In [1]:
import pandas as pd
college = pd.read_csv("../../college.csv", index_col="INSTNM")
# college
nc = college[college["STABBR"]=='NC']
# nc

# 1a) Conditions:
- Undergraduates > 7000

- Sorted descendingly by % receiving federal loans

## Question: How many of the top 50 are located in NC?

Step-wise Process: I conversed with Dr. Whaley to grab the conditions from all the colleges then pull the North Carolinian colleges from that subgroup.

In [2]:
plus7k = college[college["UGDS"]>=7000] # First condition
# plus7k.columns
sorted7k = plus7k.sort_values("PCTFLOAN",ascending=False).head(50) # Second condition 
ncplus7k = sorted7k[sorted7k["STABBR"] == 'NC']
ncplus7k # Only one school from the top 50 is located in NC

CITY STABBR  HBCU  MENONLY  \
INSTNM                                                                    
North Carolina A & T State University  Greensboro     NC   1.0      0.0   

                                       WOMENONLY  RELAFFIL  SATVRMID  \
INSTNM                                                                 
North Carolina A & T State University        0.0         0     450.0   

                                       SATMTMID  DISTANCEONLY    UGDS  ...  \
INSTNM                                                                 ...   
North Carolina A & T State University     460.0           0.0  8894.0  ...   

                                       UGDS_2MOR  UGDS_NRA  UGDS_UNKN  \
INSTNM                                                                  
North Carolina A & T State University     0.0282    0.0217     0.0281   

                                       PPTUG_EF  CURROPER  PCTPELL  PCTFLOAN  \
INSTNM                                                                         
North Carolina A & T State University    0.0651         1   0.6091    0.7947   

                                       UG25ABV  MD_EARN_WNE_P10  \
INSTNM                                                            
North Carolina A & T State University   0.1128            33000   

                                       GRAD_DEBT_MDN_SUPP  
INSTNM                                                     
North Carolina A & T State University               28500  

[1 rows x 26 columns]

### Only 1 North Carolinian college is in the top 50 category.

# 1b) Conditions:
- Undergraduates > 7000

- Sorted descendingly by % receiving federal loans

## Question: How many of the top 50 are distance only?

Step-wise Process: Similar to last problems except for the last distance condition.

In [3]:
plus7k = college[college["UGDS"]>=7000] # First condition
sorted7k = plus7k.sort_values("PCTFLOAN",ascending=False).head(50) # Second condition 
dist = sorted7k[sorted7k["DISTANCEONLY"]==1.0] # Distance only is True/1.0
dist

CITY STABBR  \
INSTNM                                                                     
American InterContinental University-Online            Schaumburg     IL   
Colorado Technical University-Online             Colorado Springs     CO   
Capella University                                    Minneapolis     MN   
The Art Institute of Pittsburgh-Online Division        Pittsburgh     PA   
Walden University                                     Minneapolis     MN   

                                                 HBCU  MENONLY  WOMENONLY  \
INSTNM                                                                      
American InterContinental University-Online       0.0      0.0        0.0   
Colorado Technical University-Online              0.0      0.0        0.0   
Capella University                                0.0      0.0        0.0   
The Art Institute of Pittsburgh-Online Division   0.0      0.0        0.0   
Walden University                                 0.0      0.0        0.0   

                                                 RELAFFIL  SATVRMID  SATMTMID  \
INSTNM                                                                          
American InterContinental University-Online             0       NaN       NaN   
Colorado Technical University-Online                    0       NaN       NaN   
Capella University                                      0       NaN       NaN   
The Art Institute of Pittsburgh-Online Division         0       NaN       NaN   
Walden University                                       0       NaN       NaN   

                                                 DISTANCEONLY     UGDS  ...  \
INSTNM                                                                  ...   
American InterContinental University-Online               1.0  10542.0  ...   
Colorado Technical University-Online                      1.0  18566.0  ...   
Capella University                                        1.0   8738.0  ...   
The Art Institute of Pittsburgh-Online Division           1.0   9374.0  ...   
Walden University                                         1.0   8658.0  ...   

                                                 UGDS_2MOR  UGDS_NRA  \
INSTNM                                                                 
American InterContinental University-Online         0.0313    0.0007   
Colorado Technical University-Online                0.0371    0.0005   
Capella University                                  0.0225    0.0029   
The Art Institute of Pittsburgh-Online Division     0.0000    0.0000   
Walden University                                   0.0216    0.0099   

                                                 UGDS_UNKN  PPTUG_EF  \
INSTNM                                                                 
American InterContinental University-Online         0.0953    0.4730   
Colorado Technical University-Online                0.0670    0.4655   
Capella University                                  0.1634    0.8143   
The Art Institute of Pittsburgh-Online Division     0.0578    0.7068   
Walden University                                   0.1499    0.9061   

                                                 CURROPER  PCTPELL  PCTFLOAN  \
INSTNM                                                                         
American InterContinental University-Online             1   0.7170    0.8922   
Colorado Technical University-Online                    1   0.6901    0.8829   
Capella University                                      1   0.5505    0.8465   
The Art Institute of Pittsburgh-Online Division         1   0.6743    0.6989   
Walden University                                       1   0.4650    0.6614   

                                                 UG25ABV  MD_EARN_WNE_P10  \
INSTNM                                                                      
American InterContinental University-Online       0.8666            40700   
Colorado Technical University-Online              0.8769        

### 5 Colleges in the top 50 are distance only.

# 1c) Conditions:
- Undergraduates > 7000

- Sorted descendingly by % receiving federal loans

## Question: What is the mean earning of students 10years after enrollment of those schools?

Step-wise Process: Similar to last problems except for removing the exceptions and calculating the mean earnings.

In [4]:
plus7k = college[college["UGDS"]>=7000] # First condition
sorted7k = plus7k.sort_values("PCTFLOAN",ascending=False).head(50) # Second condition 
# sorted7k["MD_EARN_WNE_P10"]!= "PrivacySuppressed"
dropped = sorted7k[sorted7k["MD_EARN_WNE_P10"]!= "PrivacySuppressed"] # Drop the private numbers
dropped = dropped.dropna(subset=["MD_EARN_WNE_P10"]) # Drop the NaN values
earnings = dropped["MD_EARN_WNE_P10"].astype(float).mean() # Finally take the mean
earnings

40027.083333333336

### $40027.083333333336 is the mean earning from the top 50 schools.

# 2) What is the national median debt for graduates (GRAD_DEBT_MDN_SUPP) compared to the median for NC schools?

Step-wise Process: Get rid of the private and NaN values for all colleges, then for NC.

In [5]:
dropped = college[college["GRAD_DEBT_MDN_SUPP"]!= "PrivacySuppressed"] # Drop the private numbers
dropped = dropped.dropna(subset=["GRAD_DEBT_MDN_SUPP"]) # Drop the NaN values
natMeanDebt = dropped["GRAD_DEBT_MDN_SUPP"].astype(float).median()
nc = college[college["STABBR"] == "NC"]
droppednc = nc[nc["GRAD_DEBT_MDN_SUPP"]!= "PrivacySuppressed"] # Drop the private numbers
droppednc = droppednc.dropna(subset=["GRAD_DEBT_MDN_SUPP"]) # Drop the NaN values
ncMeanDebt = droppednc["GRAD_DEBT_MDN_SUPP"].astype(float).median()
print("The national median debt is \n"+ str(natMeanDebt)+ "\nThe NC median debt is \n"+ str(ncMeanDebt))

The national median debt is 
14500.0
The NC median debt is 
19896.0


The North Carolinian median debt is higher than the national median debt, somehow.